frontend/document/picdocument.lua

```diff
@@ -63,6 +63,7 @@ function PicDocument:register(registry)
     registry:addProvider("jpg", "image/jpeg", self, 80)
     registry:addProvider("jpeg", "image/jpeg", self, 80)
     registry:addProvider("png", "image/png", self, 80)
+    registry:addProvider("webp", "image/webp", self, 80)
```

Comment:
Given that you patched MuPDF you can also add this there, presumably with a slightly lower weight. Or does it only work in CBZ?

Reply:
It looks like it doesn't work:
```
08/03/22-16:15:51 INFO  opening file test/webp-sample1.webp
ffi.load: libs/liblept.so.5
ffi.load: libs/libk2pdfopt.so.2
error: cannot recognize version marker
warning: trying to repair broken xref
warning: repairing PDF document
error: no objects found
08/03/22-16:15:52 WARN  cannot open document test/webp-sample1.webp frontend/document/pdfdocument.lua:35: ./ffi/mupdf.lua:78: MuPDF cannot open file.: no objects found (2)
```
The upstream patch probably didn't add the needed stuff to handle webp as the initial document - but only when extracting images from another type of document.
(I'm not interested enough in MuPDF to look at adding that :)

I don't really know in which contexts it works: at least when in CBZ, and when asking MuPDF to render images like we do in frontend (and in that case, it fails with an animated webp - the upstream patch doesn't use the same libwebp objects I ended up using in ffi/webp.lua and crengine - but it should be fine with non-animated webp).

Reply:
Alright, fair enough.